api/api/versions_api.go

```diff
@@ -141,6 +142,124 @@ func (c *VersionsController) CreateVersion(r *http.Request, vars map[string]stri
 	return Created(version)
 }
 
+func (c *VersionsController) DeleteVersion(r *http.Request, vars map[string]string, _ interface{}) *Response {
+	ctx := r.Context()
+
+	modelID, err := models.ParseID(vars["model_id"])
+	if err != nil {
+		return BadRequest("Unable to parse model_id")
+	}
+	versionID, err := models.ParseID(vars["version_id"])
+	if err != nil {
+		return BadRequest("Unable to parse version_id")
+	}
+
+	model, version, err := c.getModelAndVersion(ctx, modelID, versionID)
+	if err != nil {
+		if !gorm.IsRecordNotFoundError(err) {
+			return InternalServerError(err.Error())
+		}
+		return NotFound(err.Error())
+	}
+
+	// handle for pyfunc v2, since the prediction job feature is only available for pyfunc_v2 model
+	if model.Type == "pyfunc_v2" {
+		// check active prediction job
+		// if there are any active prediction job using the model version, deletion of the model version are prohibited
+		inactiveJobs, response := c.getInactivePredictionJobsForDeletion(ctx, model, version)
+		if response != nil {
+			return response
+		}
+
+		// delete inactive prediction job
+		response = c.deletePredictionJobs(ctx, inactiveJobs, model, version)
+		if response != nil {
+			return response
+		}
+	}
+
+	// check active endpoints for all model
+	// if there are any active endpoint using the model version, deletion of the model version are prohibited
+	inactiveEndpoints, response := c.getInactiveEndpointsForDeletion(ctx, model, version)
+	if response != nil {
+		return response
+	}
+	// delete inactive endpoint
+	response = c.deleteVersionEndpoints(inactiveEndpoints, version)
+	if response != nil {
+		return response
+	}
+
+	// delete mlflow run and artifact
+	err = c.MlflowDeleteService.DeleteRun(ctx, version.RunID, version.ArtifactURI, true)
+	if err != nil {
+		return InternalServerError(fmt.Sprintf("Delete Failed: %s", err.Error()))
```

Comment:
Same comments in this file, as in the models API file, about the error messages.